src/util/BaseRestClient.ts

```diff
@@ -122,6 +124,16 @@ export default abstract class BaseRestClient {
       ...restOptions,
     };
 
+    if (this.options.proxy_host){
+      this.proxy = {
+        protocol: this.options.proxy_protocol!,
+        host: this.options.proxy_host,
+        port: this.options.proxy_port!
+      };
+    } else {
+      this.proxy = undefined;
+    }
```

Comment:
Thanks for the PR & sorry for the delay. It's actually possible for you to inject these axios-parameters without any code change in the BaseRestClient. Take a look at the second parameter in the constructor of the BaseRestClient (line 108). `networkOptions` is of the type `AxiosRequestConfig`. This means you can set your axios-compatible configuration when making an instance of the rest client you're using. Example:
```typescript
const key = process.env.API_KEY_COM;
const secret = process.env.API_SECRET_COM;

const restClientOpts: RestClientOptions = {
  key: key,
  secret: secret,
};

const axiosOpts: AxiosRequestConfig = {
  proxy: {
    protocol: 'https',
    host: 'yourproxyhost',
    port: 5432
  }
};

const client = new RestClientV5(restClientOpts, axiosOpts);
```

No code change needed to support this, already works with the latest version of all my SDKs.

Reply:
I feel so silly for missing the option to pass these axiosOpts to the constructor..
Thank you for the review and example!

Reply:
Don't worry at all, it's probably not obvious enough and none of the examples use it! Either way I appreciate that you took the time to propose a PR to meet your goal - most people just ask and don't try to implement it, so I hugely appreciate your initiative 🙂

Reply:
While we are at it - have you ever tried to use a proxy for websockets (isomorphic-ws)?
I noticed it was possible to pass to bybit-api's websocketClient an agent with requestOptions.
Two options I tried are:
* https://www.npmjs.com/package/https-proxy-agent
* https://www.npmjs.com/package/better-https-proxy-agent
I tried both https/http protocol for the proxy agent.
I've tried multiple configurations for the proxy servers:
* nginx
* socat (simple tcp tunneling, listening on https)
* websocat (https://github.com/vi/websocat)

with no luck, I think most of the errors I ran into have to do with SSL handshakes failing..

Currently I've managed to get it working only in one way -
1. Using websocat as the proxy server (listening on ws:// and forwarding to wss:// on bybit)
2. it doesn't work with any of the above proxy agents, but it works if I patch the bybit-api to send requests to the proxy server on ws://

`     linear: {
         public: {
-            livenet: 'wss://stream.bybit.com/realtime_public',
-            livenet2: 'wss://stream.bytick.com/realtime_public',
-            testnet: 'wss://stream-testnet.bybit.com/realtime_public',
+            livenet: 'ws://127.0.0.1:8082/realtime_public',
+            livenet2: 'ws://127.0.0.1:8082/realtime_public',
+            testnet: 'ws://127.0.0.1:8082/realtime_public',
`

Reply:
Oh and here's how the getHttpsProxyAgent uses that https-proxy-agent npm module: https://github.com/tiagosiebler/binance/blob/master/src/util/proxy.ts#L6

Reply:
In this case it was with brightdata (used to be called luminati). Haven't tried hosting my own proxy in a long time...would be curious though about tools for it.

Reply:
As I said, I was am trying to use socat / websocat / nginx as hosted proxy servers, which work for me when I use the proxy for REST API but had no luck getting them to work with web sockets by pass the agent parameter in request options. I keep getting SSL handshake issues. 

**However**, I reviewed the code some more, and I see that **I can still manage without patching the bybit-api code**. I can just override the wsUrl parameter, in the options for the websocketClient constructor. That'll allow me to use the working solution I mentioned in my previous comment (without using the https proxy agent).